cps/tasks/metadata_extract.py

```diff
@@ -160,8 +160,14 @@ def run(self, worker_thread):
             self._remove_shorts_from_db(conn)
             requested_urls = self._fetch_requested_urls(conn)
             if not requested_urls:
-                self.message = f"{self.media_url_link} failed: Video not available."
-                conn.execute("DELETE FROM media WHERE path = ?", (self.media_url,))
+                if self.unavailable:
+                    self.message = f"{self.media_url_link} failed: Video not available."
+                if error_message := conn.execute("SELECT error FROM media WHERE ? LIKE '%' || extractor_id || '%'", (self.media_url,)).fetchone()[0]:
```

Comment:
Should `elif` be considered?

Ideally with an `else` clause at bottom, for more thorough ("paranoid") error messaging in all cases?!

```suggestion
                elif error_message := conn.execute("SELECT error FROM media WHERE ? LIKE '%' || extractor_id || '%'", (self.media_url,)).fetchone()[0]:
```